### ISSUE TITLE
"ask-sdk-local-debug package not found" when running debugger. Packages are installed.

### ISSUE THREAD

**Issue by talkingnews** [alexa/ask-toolkit-for-vscode#99]
**Describe the bug**

When running local debugger, ASK Toolkit output warns 
 `ask-sdk-local-debug package not found. Please install the ask-sdk-local-debug package from npm. `

**To Reproduce**

```
cd <skill lambda>
npm install --save ask-sdk-model@^1.29.0
npm install --save-dev ask-sdk-local-debug
```

Follow this to install launcher:
https://developer.amazon.com/en-US/docs/alexa/ask-toolkit/vs-code-testing-simulator.html#prepare

Open skill in its own workspace. All folders appear present and correct

```
.ask
.git
.gitignore
.vscode
ask-resources.json
lambda
node_modules
package-lock.json
package.json
skill-package
```

I have tried being in the lambda folder and the top level folder. I have also tried creating a package.json at the package root (above lambda) and npm install-ing model and debug at that level too. It still thinks ask-sdk-local-debug package is not installed. Debug won't run, so it'll only ever debug using the remote lambda, which defeats the point.

**Extension Logs**

```
[info - 2021-02-07 18:12:34.960Z] Activating extension 
[debug - 2021-02-07 18:12:34.962Z] Registering commands in the extension 
[debug - 2021-02-07 18:12:34.966Z] Registering views in the extension 
[debug - 2021-02-07 18:12:34.997Z] Calling method: ask.welcome 
[debug - 2021-02-07 18:12:34.998Z] Calling method: welcomeScreen.getHtmlForView 
[verbose - 2021-02-07 18:12:34.998Z] Calling method: checkGitInstallation 
[verbose - 2021-02-07 18:12:34.998Z] Calling method: isGitInstalled 
[verbose - 2021-02-07 18:12:34.998Z] Executing git command : git --version 
[debug - 2021-02-07 18:12:35.007Z] Calling method: renderView, args:  [{"name":"welcomeScreen","js":false,"args":{"enabled":"checked"}}]
[debug - 2021-02-07 18:12:35.009Z] Calling method: interpolate, args:  [{"enabled":"checked"}]
[debug - 2021-02-07 18:12:35.011Z] Registering event handlers in the extension 
[debug - 2021-02-07 18:12:35.012Z] Registering statusbar in the extension 
[verbose - 2021-02-07 18:12:35.013Z] Updating profile icon in the extension 
[debug - 2021-02-07 18:12:35.013Z] Registering treeviews in the extension 
[verbose - 2021-02-07 18:12:35.019Z] Calling method: findSkillFolders 
[verbose - 2021-02-07 18:12:35.418Z] Calling method: getOrInstantiateGitApi 
[verbose - 2021-02-07 18:12:35.420Z] Calling method: findSkillFolders 
[debug - 2021-02-07 18:12:35.421Z] Checking if extension version is latest 
[debug - 2021-02-07 18:12:39.025Z] Calling method: ask.debugAdapterPath, args: [{"name":"Debug Alexa Skill (Node.js)","type":"node","request":"launch","program":"${command:ask.debugAdapterPath}","args":["--accessToken","${command:ask.accessToken}","--skillId","${command:ask.skillIdFromWorkspace}","--handlerName","handler","--skillEntryFile","/home/ubuntu/scripts/alexa/MyTalkingNewspaperV7/lambda/index.js","--region","EU"],"cwd":"/home/ubuntu/scripts/alexa/MyTalkingNewspaperV7/lambda","__configurationTarget":5,"__debuggablePatterns":["*.js","*.es6","*.jsx","*.mjs"],"__workspaceFolder":"/home/ubuntu/scripts/alexa/MyTalkingNewspaperV7"}]
[error - 2021-02-07 18:12:39.330Z] ask-sdk-local-debug package not found. Please install the ask-sdk-local-debug package from npm. 

[info - 2021-02-07 18:12:39.856Z] skill package schema has been updated. 
[verbose - 2021-02-07 18:12:39.858Z] Calling method: getSkillMetadataSrc 
[verbose - 2021-02-07 18:12:39.858Z] Calling method: getAskResourceConfig 
[info - 2021-02-07 18:12:39.860Z] SkillPackageWatcher is started 
[debug - 2021-02-07 18:12:44.202Z] Failed to upload telemetry data. 
[debug - 2021-02-07 18:12:48.473Z] Failed to upload telemetry data. 
```

**Expected behavior**

## Your Environment

* ASK SDK for Node.js used: 2.22.4
* Operating System and version: Ubuntu 20.04

## Node.js and NPM Info
* Node.js version used for development: 14.15.4
* NPM version used for development:  7.5.2

## launch.json

```
{
    // Use IntelliSense to learn about possible attributes.
    // Hover to view descriptions of existing attributes.
    // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Debug Alexa Skill (Node.js)",
            "type": "node",
            "request": "launch",
            "program": "${command:ask.debugAdapterPath}",
            "args": [
                "--accessToken",
                "${command:ask.accessToken}",
                "--skillId",
                "${command:ask.skillIdFromWorkspace}",
                "--handlerName",
                "handler",
                "--skillEntryFile",
                "${workspaceFolder}/lambda/index.js",
                "--region",
                "EU"
            ],
            "cwd": "${workspaceFolder}/lambda"
        },
        {
            "type": "pwa-node",
            "request": "launch",
            "name": "Launch Program",
            "skipFiles": [
                "<node_internals>/**"
            ],
            "program": "${file}"
        }
    ]
}
```

![image](https://user-images.githubusercontent.com/1552808/107155678-a24b5900-6971-11eb-8adb-359340c518a6.png)

**Comment by sattpat** [comment]
@talkingnews , my hypothesis is that the node_modules folder in the project root is resulting in the `ask.debugAdapterPath` command to look for `ask-sdk-local-debug` package in the wrong folder. So lets start out with just deleting the node_modules, package.json and package-lock.json files at the root folder.

**Comment by talkingnews** [comment]
Hi @sattpat . Thank you for your good suggestion.

However, at first I did not have that extra package.json and node modules folder in the root. I only added that after it wasn't working. I can also confirm that the the node packages ar4 installed correctly in the modules folder

**Comment by sattpat** [comment]
This is very interesting. Would it be possible for you to share the entire skill package as a zip?

**Comment by talkingnews** [comment]
Sure. In fact, I used ASK Toolkit for VSCode to create a brand new template skill and the result is the same.
I've zipped it complete with the node_modules so you can see the debugger installed correctly.
https://www.dropbox.com/s/c1859de4ka6vkab/VisualStudioTest.zip?dl=1

Many thanks.

**Comment by talkingnews** [comment]
Hi @sattpat - I have found the solution. In launch.json, change

`"program": "${command:ask.debugAdapterPath}",`
to
`"program": "node_modules/ask-sdk-local-debug/dist/LocalDebuggerInvoker.js",`

Now it works.

I am using remote development on VSCode - is it possible that the Toolkit has problems reading the debugAdapterPath variable?

**Comment by sattpat** [comment]
So the command `ask.debugAdapterPath` is supposed to resolve the path to the local debug packages. Its interesting that the other vscode commands are working. Are you running this in a windows linux subsystem?

**Comment by talkingnews** [comment]
Hi @sattpat . Yes, it's very strange but I checked and double checked.

No wsl, this is just a straight Ubuntu 20.04 dev box.

**Comment by sattpat** [comment]
Gotcha and just so I understand this correct, you are running the extension on a remote machine?

**Comment by talkingnews** [comment]
Correct. So, I am running vscode locally on windows, but the toolkit extension is running on the remote Linux host via the vscode remote extension. please let me know if I can send you any logs or screenshots that might help. I can even do a short video showing me toggling between the two bits of code and the log window output. In all other respects the toolkit works ok. It is just this one environment variable

**Comment by sattpat** [comment]
Thanks @talkingnews . I will try to reproduce this issue and get back to you some time next week.

**Comment by mengtongun** [comment]
![image](https://user-images.githubusercontent.com/60088954/125924297-63f9a5bb-2042-4c04-9a4d-7c3f6db2bf8d.png)

Please help!

**Comment by Shreyas-vgr** [comment]
Created a fix to resolves this issue: PR #147
Will merge the changes soon.

**Comment by mengtongun** [comment]
> > ![image](https://user-images.githubusercontent.com/60088954/125924297-63f9a5bb-2042-4c04-9a4d-7c3f6db2bf8d.png)
> > Please help!
> 
> Hi @mengtongun, thanks for the sharing your screenshot. But I feel like this is a separate issue and not related to the current one.
> Do you mind creating a new issue along with detailed information provided in the template and also steps to reproduce ?

Actually I changed to Ubuntu and everything is working fine for me now.
